### PR TITLE
Proposed Javascript refactor

### PR DESCRIPTION
**Pull request by kdid**
Proposed changes- 

- refactored the JS a bit for readability
- added a test for a failure case
- Moved the "s3://..." checking to the Elixir code.
  - It seemed like if the source url is wrong, don't spawn the process at all. 
  - It didn't seem necessary to check the target at all since we're generating it right there. But easy to add...

### CODE REVIEW

File: priv/tiff/pyramid.js [hunk 1]
```diff
@@ -1,99 +1,70 @@
...
+      compression: "jpeg",
+      quality: 75,
+      tile: true,
+      tileHeight: 256,
+      tileWidth: 256,
+      pyramid: true
+    })
+    .toBuffer();
+  await sendToDestination(pyramidTiff, dest);
+};
```
Comment (adamjarling) [inline priv/tiff/pyramid.js]: Nice refactor.

Comment (kdid) [reply priv/tiff/pyramid.js]: Thanks @adamjarling - I just noticed I missed the `finally` clause that unlinks the file over to the left on line 39. Looking for where to stick that in...

**Comment by mbklein** [comment]
Looks good. Way better than its nodejs 8.x ancestor. :smile:

I'd like to propose doing away with all the `console` logging – it's going to clutter our logs and our test suite without adding much. If we want to log status, maybe we can find a way to send it back through the port and let Elixir log it properly?

File: priv/tiff/pyramid.js (hunk 1 above)
Comment (mbklein) [reply priv/tiff/pyramid.js]: Can you just wrap lines 9-26 in a `try { }` and then add a `finally { fs.unlink(inputFile) }` at the end?

**Pull request merged**